runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -116,6 +116,80 @@ void J9::RecognizedCallTransformer::process_java_lang_StrictMath_and_Math_sqrt(T
 
    TR::TransformUtil::removeTree(comp(), treetop);
    }
+
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+/*
+Transforms calls to java/lang/invoke/Invokers.checkExactType to the more performant ZEROCHK.
+
+Blocks before transformation: ==>
+
+start Block_A
+...
+treetop
+        call  java/lang/invoke/Invokers.checkExactType(Ljava/lang/invoke/MethodHandle;Ljava/lang/invoke/MethodType;)
+           ==>aload
+           ==>aload
+...
+end Block_A
+
+Blocks after transformation: ==>
+
+start Block_A
+...
+treetop
+        ZEROCHK
+          acmpeq
+             ==>aload <expected type>
+             aloadi <MethodHandle.type>
+               ==>aload <MethodHandle>
+...
+end Block_A
+
+*/
+void J9::RecognizedCallTransformer::process_java_lang_invoke_Invokers_checkExactType(TR::TreeTop* treetop, TR::Node* node)
+   {
+   TR::Node * methodHandleNode = node->getChild(0);
+   TR::Node * expectedTypeNode = node->getChild(1);
+   TR_J9VMBase* fej9 = static_cast<TR_J9VMBase*>(comp()->fe());
+   uint32_t typeOffset = fej9->getInstanceFieldOffsetIncludingHeader("Ljava/lang/invoke/MethodHandle;", "type", "Ljava/lang/invoke/MethodType;", comp()->getCurrentMethod());
+   TR::SymbolReference *typeSymRef = comp()->getSymRefTab()->findOrFabricateShadowSymbol(comp()->getMethodSymbol(),
+                                                                                         TR::Symbol::Java_lang_invoke_MethodHandle_type,
+                                                                                         TR::Address,
+                                                                                         typeOffset,
+                                                                                         false,
+                                                                                         false,
+                                                                                         true,
+                                                                                         "java/lang/invoke/MethodHandle.type Ljava/lang/invoke/MethodType;");
+   TR::Node *handleTypeNode = TR::Node::createWithSymRef(comp()->il.opCodeForIndirectLoad(TR::Address), 1, 1, methodHandleNode, typeSymRef);
```

Comment:
Nodes created after ilgen should have originating bytecode node. Use the node create API that takes a node as the first argument to create the node.